tmc-langs-util/src/main/java/fi/helsinki/cs/tmc/langs/util/ProjectTypeHandler.java

```diff
@@ -0,0 +1,52 @@
+package fi.helsinki.cs.tmc.langs.util;
+
+
+import fi.helsinki.cs.tmc.langs.LanguagePlugin;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+
+public class ProjectTypeHandler {
+
+    private final HashMap<ProjectType, LanguagePlugin> projectTypes = new HashMap<>();
+
+    /**
+     * Register all the available LanguagePlugins to ProjectTypeHandler
+     */
+    public ProjectTypeHandler() {
+        for(ProjectType type : ProjectType.values()) {
+            projectTypes.put(type, type.getLanguagePlugin());
+        }
+    }
+
+    /**
+     * Recognise the project type.
+     *
+     * <p>
+     * Iterate through all language plugins to find one that recognises the
+     * project as their type.
+     *
+     * @param path The path to the exercise directory.
+     * @return The project type, or null if none.
+     */
+    public ProjectType getProjectType(Path path) {
+        for(ProjectType type : projectTypes.keySet()) {
```

Comment:
The hashmap seems unnecessary. You could just iterate over ProjectType.values() and get the plugin from the ProjectType.

Also, our coding style says there should be a space after keywords like `for` and `if` (but not after method names).